pkg/chunk/table_manager.go

```diff
@@ -205,20 +204,31 @@ func (m *TableManager) SyncTables(ctx context.Context) error {
 	expected := m.calculateExpectedTables()
 	level.Info(util.Logger).Log("msg", "synching tables", "num_expected_tables", len(expected), "expected_tables", len(expected))
 
-	toCreate, toCheckThroughput, toDelete, err := m.partitionTables(ctx, expected)
-	if err != nil {
-		return err
-	}
+	for i, client := range m.clients {
+		toCreate, toCheckThroughput, toDelete, err := m.partitionTables(ctx, client, expected)
```

Comment:
This doesn't seem right. This would mean that every single client would create a bunch of random tables (ones that belong to other clients).

Can we pass `config` to `calculateExpectedTables()` and then use that for each of the clients?

Reply:
I was thinking the same. But it is the same behavior before apparently. `calculateExpectedTables` parses all the configs, and creates/updates/deletes from the last client. Here I was only deleting from old client, but left the behavior of the last client unchanged.

Do you want me to change it to per client for `calculateExpectedTables`?

Reply:
But the delete calls will fail, as they are trying to delete tables that don't exist. Yeah, changing `calculateExpectedTables` makes sense.

Reply:
Tests failed for the change described. While I am still digging into the test failure, from past implementation my understanding is that `i th` client uses all config from the starting till the `i th` config to create tables.

Reply:
I changed the implementation to use `0,1,...,i'th` config for the `i'th` client. Tests pass for that. It that the expected behaviour then?

Reply:
Doesn't sound right.  Why wouldn't the i'th client just use the i'th config?

Reply:
What you said sounds right. Apparently the old code used all the configs for the client. I will update the tests then.

Reply:
Now I see why all the configs were parsed in the old code
https://github.com/cortexproject/cortex/blob/53438dfee1beec34876b14fc977f345158c351dc/pkg/chunk/table_manager.go#L244-L255

The old configs are parsed and checked for grace period w.r.t new configs. Also provisions read/write accordingly if the config is active/inactive. I guess we would need parsing the old configs if we want to delete the tables from the client if tables were created because of the grace period. So `0,1,...,i'th` configs are only parsed inside `calculateExpectedTables` for the `i'th` client, and not necessarily used.

Reply:
Could we make a rule that grace period only applies to the latest config?

Reply:
Done.